### DISCUSSION TITLE
Can Flock Revolutionize Web-Based Rendering and Replace HTML?

### DISCUSSION THREAD

**Discussion by AnuragCsharp** [join-the-flock/nest#23] [Q&A]
Hello everyone,

I’ve been exploring the potential of Flutter, and its versatility for building cross-platform applications has been truly impressive. With Flutter for Web, it seems like we’re moving closer to a world where the traditional reliance on HTML for web rendering might be challenged.

Here’s where my curiosity kicks in:

Can Flutter's Skia-based rendering engine and its pixel-perfect design capabilities eventually make it a viable alternative to HTML for web-based UI rendering?
What are the potential advantages of adopting Flutter over HTML/CSS/JS for modern web applications?
Could this shift improve web development performance, maintainability, or the developer experience as a whole?
While HTML has been the cornerstone of web rendering for decades, frameworks like Flutter seem to offer a new way of thinking. Do you think this could lead to a paradigm shift, or is HTML too deeply entrenched in the web ecosystem to be replaced?

Looking forward to hearing your thoughts and engaging in a vibrant discussion!

**Comment by a-man-called-q** [comment]
I'm not an expert on flutter but this is my opinion.

Short answer: No.

Long Answer:

- Flutter compiles to HTML, JS, and CSS for web.
- Flutter is a framework, HTML is a language.
- Skia is for 2D rendering while HTML is for well.. text.

If you're looking for cross platfrom that relies heavily on 2D rendering and canvas then flutter is one of the better options for the current solution. But if you're looking for CRUD stuff then there are lots of options out there from many available frameworks.

For example, I rather use flutter rather than react native if the requirement is Desktop - Mobile - Web with the same layouting, capabilities, and theme. However if the requirements are different. Then I rather use multiple frameworks.

**Comment by jezell** [comment]
@AnuragCsharp technologies like Flutter are definitely going to cause a shift away from HTML for apps. The primary advantage at the moment is that you get a much higher quality cross platform app out of the process than using HTML. You can also create cross platform apps with HTML + CSS + JS, but the browser doesn't offer enough control to close the gap between web and native apps. There is no real electron style solution for mobile due restrictions from the mobile platforms.

**Reply by a-man-called-q** [reply]
While I do agree for you, it seems like you missed his point tho. He's asking if **Flutter will replace HTML in web** which *not a chance*.

**Comment by matthew-carroll** [comment]
This conversation seems to be focused on topics that are unrelated to Flock and Nest. Therefore, I'm going to close this Discussion. If you'd like to continue this, then you might want to follow up on X, or Bluesky, or the new Flutter forum.
